### DISCUSSION TITLE
ZSTD?

### DISCUSSION THREAD

**Discussion by beckend** [nonzzz/vite-plugin-compression#61] [Q&A] [answered]
# Feature request 🚀

![image](https://github.com/user-attachments/assets/da32d56d-f296-4bec-af49-b4147d05183c)

Chromium

**Comment by nonzzz** [comment]
Any others? Aka, Currently plugin support custom  algorithm

**Comment by nonzzz** [comment, marked as answer]
Here is a simple example for zstd usage.

- setup option for compression plugin.

```ts

import { compress } from '@mongodb-js/zstd'
import { compression } from 'vite-plugin-compression2'

defineConfig({
  plugins: [
  // .... your vite's plugins
    compression < { level: number } > ({
      algorithm(buf, opt) {
        return compress(buf as any, opt)
      },
      compressionOptions: {
        level: 9
      },
      filename: '[path][base].zst'
    })
  ]
})

```
 then run vite build

**Reply by jclaveau** [reply]
Thank you! I personally had to wright it this way

```js
        compression({
          // https://github.com/nonzzz/vite-plugin-compression/discussions/61#discussioncomment-10243200
          algorithm(buf, level) {
            return compress(buf, level)
          },
          level: 9,
          filename: `[path][base].zst`,
        }),        
```

**Comment by nonzzz** [comment]
Here is a simple usage. Can you have a try?

**Comment by beckend** [comment]
I get .gz extensions, not sure if that is expected, the files are compressed.

And also type errors.

![image](https://github.com/user-attachments/assets/4cba6059-bf77-4168-9d7a-59e88a33db25)

**Comment by nonzzz** [comment]
Oh, you should set filename option :)

**Comment by beckend** [comment]
Then it works ok, the types are a bit misleading.

**Comment by CheesyTech** [comment]
Hi @nonzzz , @beckend  . Could you share a complete example of zstd integration for this plugin? 
It would also be great if someone would immediately give examples of compression integration for this library )

**Comment by nonzzz** [comment]
@cheeeasy2501  I'll move this issue to discussion

**Comment by nonzzz** [comment]
@cheeeasy2501 Already created an answer. Check the discussions pls. Or you have any problem can see [normal example](https://github.com/nonzzz/vite-plugin-compression/tree/master/example)
